daemon/daemon_unix.go

```diff
@@ -745,9 +745,6 @@ func verifyDaemonSettings(conf *config.Config) error {
 
 // checkSystem validates platform-specific requirements
 func checkSystem() error {
-	if os.Geteuid() != 0 {
```

Comment:
Isn't  this check still needed if rootless mode is not enabled? (or is that handled in the other location I commented?)